packages/front-end/components/Forms/CodeTextArea.tsx

```diff
@@ -49,6 +53,28 @@ export default function CodeTextArea({
     otherProps.error = semicolonWarning;
   }
 
+  const theme = useMemo(() => {
+    let actualTheme = LIGHT_THEME;
+
+    try {
+      const fromStorage = localStorage.getItem("gb_ui_theme");
+      if (
+        !fromStorage &&
+        window.matchMedia("(prefers-color-scheme: dark)")?.matches
+      ) {
+        actualTheme = DARK_THEME;
+      }
+
+      if (fromStorage === "dark") {
+        actualTheme = DARK_THEME;
+      }
+    } catch (e) {
+      return actualTheme;
+    }
+
+    return actualTheme;
+  }, []);
```

Comment:
We need this same logic in the `Code` component.  It already supports light/dark mode, but the theme is passed in as a prop right now instead of using the actual selected theme.

It's currently using the "light" theme to render JSON on the feature page and the SQL/Python code for data sources:

![image](https://user-images.githubusercontent.com/1087514/193817652-445a0fc4-45ea-41b5-9042-10b4f81cace8.png)
![image](https://user-images.githubusercontent.com/1087514/193818204-40c90d12-97c9-426a-a17d-67e6080c174b.png)


And it's using the "dark" theme when rendering code samples and SQL queries on the metric page:

![image](https://user-images.githubusercontent.com/1087514/193818464-a22265be-1aac-4b70-b8a5-f56fdba5c967.png)

![image](https://user-images.githubusercontent.com/1087514/193818580-3eee4c96-efe9-4ee9-8dad-5680e6e8a902.png)

Reply:
I've updated all code blocks to use the theme from the context provider.

![CleanShot 2022-10-04 at 14 08 53@2x](https://user-images.githubusercontent.com/113377031/193929344-d2b41898-6cf7-4564-924a-1643d908feff.png)
![CleanShot 2022-10-04 at 14 09 50@2x](https://user-images.githubusercontent.com/113377031/193929570-b3f7d1c9-5ff4-42e7-af37-070d5b04fc9d.png)

<img width="399" alt="image" src="https://user-images.githubusercontent.com/113377031/193929715-7ff34ffd-f063-4ce0-8c8a-812284909ba0.png">
<img width="347" alt="image" src="https://user-images.githubusercontent.com/113377031/193929758-9c72c853-89fa-4944-85a7-3a2c72a46bb3.png">